Loop/Managers/DeviceDataManager.swift

```diff
@@ -391,10 +410,58 @@ final class DeviceDataManager: CarbStoreDelegate, CarbStoreSyncDelegate, DoseSto
                     nsPumpStatus = nil
                 }
                 self.nightscoutDataManager.uploadDeviceStatus(nsPumpStatus)
+                completion?()
             }
         }
     }
 
+    private func updatePumpCGMSensorStatus(_ events: [TimestampedGlucoseEvent]) -> Void {
```

Comment:
I think @bharat has some good ideas about further improvements, and is doing the right thing by opening separate issues to track those. I think we can get this patch merged in with just the naming and mark changes.